process_proposal.org

```diff
@@ -243,111 +288,352 @@ This is a first cut at th wording. We probably don't need to go much deeper in t
+   }
+
+   // A Pid type is an identifier for a process, that satisfies StrictTotallyOrdered
    using pid_type = implementation-defined;
+   
 
-   // Launches a process and waits for its exit
-   template<typename ...Args>
-   inline int system(Args && ...args);
-
-   //TODO Jeff: I believe we can say that process satisfies SemiRegular -- copyable and default constructable.  It isn't regular bc we can't compare process objects
+   //Process satisfies Movable & Boolean, but not Copyable.
 
    //Provides a portable handle to an operating system process
-   class process
-   {
-    public:
-      //provides access to underlying operating system facilities
-      typedef platform_specific native_handle_type; 
+   class process;
+   
+   //Provides a portable wrapper for a process group
+   class process_group;
+   
+   //Provides an initializers for the standard io. Alternative: nested as std::process::io
+   class process_io;
+   
+   //Provides a way to set the starting directory of the new process.  Alternative: nested as std::process::start_dir
+   class process_start_dir;
+
+   //Satisfies ProcessInitializer
+   class environment;
+   
+   //Satisfies ProcessInitializer
+   class process_limit_handles;
+      
+}
+#+END_SRC
+
+
+** Class ~process~
 
-      // Construct a child from a property list and launch it
-      template<typename ...Args>
-      explicit process(Args&&...args);
+#+BEGIN_SRC c++
+class process
+{
+  public:
+    //provides access to underlying operating system facilities
+    typedef platform_specific native_handle_type; 
+
+    // Construct a child from a property list and launch it.
+    template<ProcessInitializer ...Inits>
+    explicit process(const std::filesystem::path & exe, const Range<std::string> & args, Inits&&...inits);
 
-      // Attach to an existing process
-      explicit process(pid_type & pid);
+    // Construct a child from a property list and launch it with a custom process launcher
+    template<ProcessInitializer ...Inits, ProcessLauncher Launcher>
+    explicit process(const std::filesystem::path & exe, const Range<std::string> & args, Inits&&...inits, Launcher && launcher);
 
-      // TODO jeff g -- what's the use case for default construction?
-      process() = default;
 
-      process(process && lhs);
-      process& operator=(process && lhs);
+    // Attach to an existing process
+    explicit process(pid_type & pid);
+
+    // An empty process is similar to a default constructed thread. It holds an empty handle and is a place holder for a process that is to be launched later.
+    process() = default;
+
+    process(process && lhs);
+    process& operator=(process && lhs);
  
-      //TODO Klemens -- I marked this noexcept -- is that a problem?
-      // calls terminate if process is not detached or !running()
-     ~process() noexcept;
+    
+    // The destructor terminates 
```

Comment:
Assuming there was supposed to be a bit more here?

Reply:
Not really, destructors are always noexcept. So yeah, correct.